### ISSUE TITLE
WSL2 should have a command to show/move where distros are stored on the filesystem

### ISSUE THREAD

**Issue by wiwa** [microsoft/WSL#9614]
**Is your feature request related to a problem? Please describe.**
Problem: No way to easily figure out where WSL2 distros are stored, and no easy way to change their location.
Futhermore, there's no easy way to see WSL2 distro information/metadata.
`wsl --list --verbose` has no filesystem info.

Moving a distro's data is often encountered when installing Docker Desktop for Windows.
i.e. we want to move the `ext4.vhdx` to another drive with more space than `C:/`.

**Describe the solution you'd like**
Introduce two commands/options:
```bash
wsl --info <distro> # produces all metadata regarding a certain distro, i.e. where it is stored on disk, size, etc.
wsl --move <distro> <absolute path to new parent directory for distro data> # e.g. wsl --move docker-desktop-data D:/docker-desktop-data
```

**Describe alternatives you've considered**
Alternative:
`--info`: Googling it every time.  
`--move`: Googling how to do this every time, exporting and importing the disk data with `wsl --export` and `wsl --import`.

**Comment by aki-k** [comment]
Until --info is implemented:

    C:\>dir C:\users\ext4.vhdx /s

**Comment by dude0001** [comment]
Does the functionality you are looking for already exists? e.g. `wsl --export` and `wsl --import`.

These are some instructions for how to create a base image to distribute to multiple users in an enterprise https://learn.microsoft.com/en-us/windows/wsl/enterprise. And these are some instructions about exporting a distro image from Docker and importing from either the WSL distro or Docker export https://learn.microsoft.com/en-us/windows/wsl/use-custom-distro#importing-centos-example.

You should be able to use a combination of these commands to "move" the storage path for your WSL distribution.

**Comment by wiwa** [comment]
Indeed, that's the alternative I mentioned in my initial post. However, that's clearly less ergonomic and user-friendly than simply baking it into WSL.

**Comment by NotTheDr01ds** [comment]
Agreed that some additional "filesystem" query/manipulation commands would be nice additions to `wsl.exe` proper, but also throwing this PowerShell snippet out (which is what I use):

```
Get-ChildItem HKCU:\Software\Microsoft\Windows\CurrentVersion\Lxss\ |
  ForEach-Object {
    (Get-ItemProperty $_.PSPATH)
  }
```

You can, of course, wrap this in a function in your PowerShell profile:

```
function Get-WSLDistroInfo {
  Get-ChildItem HKCU:\Software\Microsoft\Windows\CurrentVersion\Lxss\ |
    ForEach-Object {
      (Get-ItemProperty $_.PSPATH)
    }
}
```
